Source/Core/Core/IOS/FS/FileSystemProxy.cpp

```diff
@@ -28,12 +28,60 @@ static IPCReply GetFSReply(s32 return_value, u64 extra_tb_ticks = 0)
   return IPCReply{return_value, (2700 + extra_tb_ticks) * SystemTimers::TIMER_RATIO};
 }
 
-/// Amount of TB ticks required for a superblock write to complete.
-constexpr u64 SUPERBLOCK_WRITE_TICKS = 3370000;
-/// Amount of TB ticks required to write a cluster (for a file).
-constexpr u64 CLUSTER_WRITE_TICKS = 300000;
-/// Amount of TB ticks required to read a cluster (for a file).
-constexpr u64 CLUSTER_READ_TICKS = 115000;
+/// Duration of a superblock write (in timebase ticks).
+constexpr u64 GetSuperblockWriteTbTicks(int ios_version)
+{
+  if (ios_version == 28 || ios_version == 80)
+    return 3350000;
+
+  if (ios_version < 28)
+    return 4100000;
+
+  return 3170000;
+}
+
+/// Duration of a file cluster write (in timebase ticks).
+constexpr u64 GetClusterWriteTbTicks(int ios_version)
+{
+  // Based on the time it takes to write two clusters (0x8000 bytes), divided by two.
+  if (ios_version < 28)
+    return 370000;
+
+  return 300000;
+}
+
+/// Duration of a file cluster read (in timebase ticks).
+constexpr u64 GetClusterReadTbTicks(int ios_version)
+{
+  if (ios_version == 28 || ios_version == 80)
```

Comment:
It seems weird that IOS 80 is an outlier here. Do we know why?

Reply:
It's not clear to me why IOS80 is slower, but it appears to use a different branch of the FS module:

```
IOS9: $IOSVersion:  FS: 12/23/08 17:11:51 64M $
IOS13: $IOSVersion:  FS: 12/23/08 17:51:08 64M $
IOS21 (same)
IOS22 (same)
IOS28: $IOSVersion:  FS: 12/24/08 12:57:39 64M $

IOS36: $IOSVersion:  FFS: 12/23/08 17:26:21 64M $

IOS55: $IOSVersion:  FFSP: 12/24/08 13:48:17 64M $
IOS56: $IOSVersion:  FFSP: 11/24/08 15:36:10 64M $
IOS59 (same)
IOS62 (same)
IOS59 (Wii U): $IOSVersion:  FFSP: 04/03/12 12:47:01 64M $

IOS80: $IOSVersion:  FSP: 11/24/08 15:36:10 64M $
```

Maybe FSP is closer to FS than to FFSP? I haven't looked at 80 so I don't know how they differ.

Reply:
IOS9 was used by the System Menu 1.0, and IOS80 is used by System Menu 4.3. Other IOSes used by the System Menu are:
* IOS11 for 2.0 and 2.1
* IOS20 for 2.2
* IOS30 for 3.0 - 3.3
* IOS50 for 3.4
* IOS60 for 4.0 and 4.1
* IOS70 for 4.2

I'm not sure if you can check those for the relevant info the same way you did for IOS9 or IOS80, but maybe it's possible that they needed to keep certain timings the same for the System Menu or something?

I'm pretty sure these aren't used by any third-party software because they are replaced with non-functioning versions on each subsequent update so would cause issues if third-party software tried to use them. I'm not sure if older versions of Nintendo's own Wii channels used them as well though.

Reply:
Oh yeah, it sounds plausible that other System Menu IOS versions would have the same timing characteristics. Because those IOSes are stubbed on an up-to-date console, they weren't tested as part of the timing tests, though.

I don't think the timing differences were introduced on purpose for the system menu, it's probably just because of the system menu IOSes all coming from one branch.